packages/message-relayer/src/service.ts

```diff
@@ -238,16 +251,26 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
           }
 
           const receipt = await this._relayMultiMessageToL1(
-            this.state.messageBuffer
+            this.state.messageBuffer.reduce((acc, cur) => {
+              acc.push(cur.payload)
+              return acc
+            }, [])
           )
 
           console.log('Receipt:', receipt)
 
           /* parse this to make sure that the mesaage was actually relayed */
-          if (/*everything went well*/ true) {
+          // clear out buffer only if the messages are relayed to L1 successfully
+          if (
+            await this._wasMessageRelayed(this.state.messageBuffer[0].message)
```

Comment:
a different scenario to the comment below could be when - checking the success status of the first message could give us a false positive, where the first message was relayed but not success, but the other messages in the batch where success.